completion/services.py

```diff
@@ -76,6 +76,29 @@ def fill_in_run(block_key):
                 completions[candidate] = 0.0
         return completions
 
+    def get_user_children(self, node):
+        """
+        Verticals will sometimes have children that also have children (some
+        examples are content libraries, split tests, conditionals, etc.). This
+        function will recurse through the children to get to the bottom leaf
+        and return only those children.
+
+        Note: The nodes passed in should have already taken the user into
+        account so the proper blocks are shown for this user.
+        """
+        if node.scope_ids.block_type != 'discussion':
+            return []
+
```

Comment:
@Dillon-Dumesnil  I was hoping a refactoring here would get rid of this special casing and instead check if the block had [a `compeltion_mode` of `XBlockCompletionMode.EXCLUDED`](https://github.com/edx/edx-platform/blob/61e1eda20df2825a409db3e2d36c69d7c36d3e2d/openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion/__init__.py#L43)

Currently, there is still a bug, in that blocks that are marked as excluded and will never have a completion score will still be counted, and any vertical with such a block will never get marked as completed. 
CC: @cpennington

Reply:
Sorry, I notice this has already been fixed. I came across this PR and figured it was the final say in the matter.

Reply:
@xitij2000 Ah yes, we realized the previous logic was not always doing as intended and Cale brought up using the EXCLUDED field. Thank you for also noticing and reaching out! This also reminded me that I forgot to update the CHANGELOG so thank you for that as well!